### PR TITLE
Refactor proxy middleware and increase coverage

### PR DESCRIPTION
**Pull request by klipitkas**
- Refactor the proxy middleware.
- Update coverage from 0% to 92%.

### CODE REVIEW

File: middleware/proxy/proxy.go [hunk 1]
```diff
@@ -58,7 +58,9 @@ func Balancer(config Config) fiber.Handler {
 		cfg.Next = ConfigDefault.Next
 	}
 	if len(cfg.Servers) == 0 {
-		panic("Servers cannot be empty")
+		return func(c *fiber.Ctx) (err error) {
+			panic("Servers cannot be empty")
+		}
```
Comment (hi019) [inline middleware/proxy/proxy.go:63]: Wouldn't it be better to keep it as it is and panic on MW creation?

File: middleware/proxy/proxy_test.go [hunk 2]
```diff
@@ -1,148 +1,161 @@
...
+func Test_Proxy_Empty_Upstream_Servers(t *testing.T) {
+	app := fiber.New()
+	app.Use(recover.New(), Balancer(Config{Servers: []string{}}))
+
+	resp, err := app.Test(httptest.NewRequest("GET", "/", nil))
+	utils.AssertEqual(t, nil, err)
+	utils.AssertEqual(t, fiber.StatusInternalServerError, resp.StatusCode)
+}
 
 // // go test -run Test_Proxy_Next
```
Comment (hi019) [inline middleware/proxy/proxy_test.go:25]: Should be ` // go test -run Test_Proxy_Next`

**Comment by hi019** [comment]
Hi :wave:

Tests were broken because localhost was used in `app.Listen`. I've committed to your repository and fixed the problem. Otherwise, nice PR!

File: middleware/proxy/proxy.go (hunk 1 above)
Comment (hi019) [inline middleware/proxy/proxy.go:63]: Shouldn't we keep the code as it is and panic on MW creation?

File: middleware/proxy/proxy_test.go (hunk 2 above)
Comment (hi019) [inline middleware/proxy/proxy_test.go:25]: Remove two forward slashes. Line 30 as well :)

File: middleware/proxy/proxy.go (hunk 1 above)
Comment (klipitkas) [reply middleware/proxy/proxy.go:63]: This has an issue where it cannot be caught by the `recover` middleware.

File: middleware/proxy/proxy_test.go (hunk 2 above)
Comment (klipitkas) [reply middleware/proxy/proxy_test.go]: we should be good here, thanks

**Pull request merged**